specs/error_state/ErrorCodeStore.md

```diff
@@ -1,54 +1,40 @@
-for returned bytecodes, store them cost additional gas.   
+```
+let CODE_DEPOSIT_BYTE_COST = 200
+code_store_cost = CODE_DEPOSIT_BYTE_COST * len(bytecodes)
+``` 
 
-`let CODE_DEPOSIT_BYTE_COST = 200
-code_store_cost = CODE_DEPOSIT_BYTE_COST * bytecodes' length
-`  
+- If `code_store_cost` > gas left, it is `CodeStoreOutOfGas` case.
+- If returned bytecode length > `MAXCODESIZE`, it is `MaxCodeSizeExceeded` case.  
 
-if `code_store_cost` > gas left, it is `CodeStoreOutOfGas` case.  
-if returned bytecodes' length > `MAXCODESIZE` allowed in evm, it is 
-`MaxCodeSizeExceeded` case.  
-
-In circuit bus mapping side, check these two code store errors in [here](https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/main/bus-mapping/src/circuit_input_builder/input_state_ref.rs#L1148&L1155)
-when executing opcode is `RETURN` and call context is creating(`call.is_create == true`) meanwhile.  
-
-Even though errors occur in `CREATE` kind opcodes, it is special not checking error 
-in executing opcode `CREATE` directly.  
-Circuit implementation takes similar strategy, not constrain error directly in CREATE opcodes, but 
-in `RETURN` step context. Following this way it easy to get the key property state `length` and construct constraints against it.
+On the circuit bus mapping side, the checks for these two code store errors are [here](https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/8a633f7c3de2da72f0817def57c1703241cced97/bus-mapping/src/circuit_input_builder/input_state_ref.rs#L1296-L1304). This error happens only when current opcode is `RETURN` and it's a `CREATE`/`CREATE2` call (`call.is_create == true`). We can't get contract bytecode length in `CREATE`/`CREATE2` opcodes and it's only available in `RETURN` opcode so we handle these two errors in `RETURN` opcode.
 
 Overall it looks like the following:  
 - Pop EVM word `offset` and `length` from the stack, 
-- Go to `ErrorCodeStore` state when call context is being created & 
-select which one of the followings occurs:
-
-1.  Storing `length` of bytecodes  runs out of gas.
-2.  `length` of bytecodes exceeds `MAXCODESIZE`.
+- Go to `ErrorCodeStore` state when call context is being created & select which one of the followings occurs:
+  1. Storing `length` of bytecode runs out of gas.
+  2. `length` of bytecode exceeds `MAXCODESIZE`.
 
 ### Constraints
-1. `code_store_cost` > gas_left or `length` > `MAXCODESIZE`
-2. Current call must be failed.
-3. Current call is not static call.
-4. If it's a root call, it transits to `EndTx`
-5. if it is not root call, it restores caller's context by reading to `rw_table`, then does step state transition to it.
+1. Current opcode is `RETURN` and `is_create` is `True`.
+2. `code_store_cost` > gas_left or `length` > `MAXCODESIZE`
+3. Current call must be failed.
+4. Current call is not static call.
+5. If it's a root call, it transits to `EndTx`
```

Comment:
```suggestion
5. If it's a root call, it transits to `EndTx`. Otherwise the caller's context is restored by reading the `rw_table`, then it does step state transition to it.
```

Reply:
I prefer not to make this change in this PR bcs these two points (5 and 6) are more like a fixed pattern and all the error state MD files using the same description. If we would like to revise them, we should do it in another refinment PR to revise all. What do you think?

Reply:
Sure!